server/src/main/java/com/hedera/block/server/consumer/LiveStreamObserverImpl.java

```diff
@@ -0,0 +1,129 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.consumer;
+
+import com.hedera.block.protos.BlockStreamServiceGrpcProto;
+import com.hedera.block.server.mediator.StreamMediator;
+import io.grpc.stub.StreamObserver;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+
+/**
+ * The LiveStreamObserverImpl class implements the LiveStreamObserver interface to pass blocks to the downstream consumer
+ * via the notify method and manage the bidirectional stream to the consumer via the onNext, onError, and onCompleted methods.
+ */
+public class LiveStreamObserverImpl implements LiveStreamObserver<BlockStreamServiceGrpcProto.Block, BlockStreamServiceGrpcProto.BlockResponse> {
+
+    private final System.Logger LOGGER = System.getLogger(getClass().getName());
+
+    private final StreamMediator<BlockStreamServiceGrpcProto.Block, BlockStreamServiceGrpcProto.BlockResponse> mediator;
+    private final StreamObserver<BlockStreamServiceGrpcProto.Block> responseStreamObserver;
+
+    private final long timeoutThresholdMillis;
+
+    private final Clock producerLivenessClock;
+    private Instant producerLivenessInstant;
+
+    private final Clock consumerLivenessClock;
+    private Instant consumerLivenessInstant;
+
+    /**
+     * Constructor for the LiveStreamObserverImpl class.
+     *
+     * @param mediator the mediator
+     * @param responseStreamObserver the response stream observer
+     */
+    public LiveStreamObserverImpl(
+            final long timeoutThresholdMillis,
+            final Clock producerLivenessClock,
+            final Clock consumerLivenessClock,
+            final StreamMediator<BlockStreamServiceGrpcProto.Block, BlockStreamServiceGrpcProto.BlockResponse> mediator,
+            final StreamObserver<BlockStreamServiceGrpcProto.Block> responseStreamObserver) {
+
+        this.timeoutThresholdMillis = timeoutThresholdMillis;
+        this.producerLivenessClock = producerLivenessClock;
+        this.consumerLivenessClock = consumerLivenessClock;
+        this.mediator = mediator;
+        this.responseStreamObserver = responseStreamObserver;
+
+        this.producerLivenessInstant = Instant.now(producerLivenessClock);
+        this.consumerLivenessInstant = Instant.now(consumerLivenessClock);
+    }
+
+    /**
+     * Pass the block to the observer provided by Helidon
+     *
+     * @param block the block to be passed to the observer
+     */
+    @Override
+    public void notify(final BlockStreamServiceGrpcProto.Block block) {
+
+        // Check if the consumer has timed out.  If so, unsubscribe the observer from the mediator.
+        if (Duration.between(consumerLivenessInstant, Instant.now(consumerLivenessClock)).toMillis() > timeoutThresholdMillis) {
```

Comment:
Here instead of `Instant.now(consumerLivenessClock)).toMillis()` you could use `consumerLivenessClock.millis()` and avoid the extra object.

Reply:
Fixed